ocaml-lsp-server/src/inferred_intf.ml

```diff
@@ -0,0 +1,48 @@
+open Import
+
+let action_kind = "inferred_intf"
+
+let infer_intf impl =
+  Document.with_pipeline impl (fun pipeline ->
+      let typer = Mpipeline.typer_result pipeline in
+      let sig_ : Types.signature =
+        let typedtree = Mtyper.get_typedtree typer in
+        match typedtree with
+        | `Interface _ ->
+          Printf.fprintf stderr "assert false\n";
+          assert false
+        | `Implementation impl -> impl.str_type
+      in
+      Format.asprintf "%a@." Printtyp.signature sig_)
```

Comment:
This call needs to be wrapped with:

```ocaml
Printtyp.wrap_printing_env env ~verbosity
```

This is done so that `-short-paths` is respected.